### PR TITLE
feat: add section for user / group provisioning

### PR DESCRIPTION
**Pull request by j0g3sc**
CU-1p69at5

### CODE REVIEW

File: docs/meshstack.user-group-provisioning.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+Provisioning users and groups to meshStack works based on the [SCIM 2.0](http://www.simplecloud.info/) standard.
+We provide an SCIM API that enables external systems such as an Azure Active Directory (AAD) to directly
+manage up-to-date user and group information within meshStack. `Provisioning` means changes about users and groups
+in the external system are automatically mirrored to the provisioned equivalents in meshStack. For example, when
+the members of a certain AAD group are changing and provisioning is set up, those changes will be reflected in the group within meshStack as well
+without additional effort. That way we can model the complete lifecycle of users and groups automatically.
+
+## Provisioning with SCIM
+
+SCIM clients need to address users and groups in order to manage the creation, updates and deletion. Because they cannot know
```
Comment (chewbeccy) [inline docs/meshstack.user-group-provisioning.md:15]: we should clarify that SCIM client are the source like AAD. I

File: docs/meshstack.user-group-provisioning.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+```
+
+## SCIM API limitations
+
+The SCIM standard introduces comprehensive functionality to work with user and group resources.
+The implementation of meshStack's SCIM API does not support all features of SCIM. Here is a list of current limitations:
+
+- Deletion of users is not supported.
+- Filtering options are reduced:
```
Comment (chewbeccy) [inline docs/meshstack.user-group-provisioning.md:79]: this has nothing to do with the scoping of the sync right? This is how the SCIM client can make changes to existing identities in meshStack?

**Review by chewbeccy — COMMENTED** [review]
Some remarks

File: docs/meshstack.user-group-provisioning.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    ]
+}
+```
+
+## SCIM API limitations
+
+The SCIM standard introduces comprehensive functionality to work with user and group resources.
+The implementation of meshStack's SCIM API does not support all features of SCIM. Here is a list of current limitations:
+
+- Deletion of users is not supported.
```
[the remark this answers is not in the document]
Comment (j0g3sc) [reply docs/meshstack.user-group-provisioning.md:78]: oh very good point, will clarify that.

File: docs/meshstack.user-group-provisioning.md (hunk 2 above)
Comment (j0g3sc) [reply docs/meshstack.user-group-provisioning.md:79]: clarified that, those are filter options and parameters that you can apply to search users and groups.

**Review by chewbeccy — APPROVED** [review]

**Pull request merged**